tools/junit-openfeature/src/main/java/dev/openfeature/contrib/tools/junitopenfeature/TestProvider.java

```diff
@@ -0,0 +1,128 @@
+package dev.openfeature.contrib.tools.junitopenfeature;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.FlagNotFoundError;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import dev.openfeature.sdk.providers.memory.Flag;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * TestProvider based on InMemoryProvider but with another dimension added to the maps of flags.
+ */
+@Slf4j
+public class TestProvider extends EventProvider {
+    public static final ThreadLocal<ExtensionContext.Namespace> CURRENT_NAMESPACE = new ThreadLocal<>();
+    public Map<ExtensionContext.Namespace, Map<String, Flag<?>>> flagsMap = new HashMap<>();
```

Comment:
This can/should be private isn't it?

Reply:
yes definitely, i experimented with a little different approach of one static map, and forgot to change the access modifier